CMakeLists.txt

```diff
@@ -161,7 +161,7 @@ if(SPDLOG_BUILD_SHARED OR BUILD_SHARED_LIBS)
     endif()
     add_library(spdlog SHARED ${SPDLOG_SRCS} ${SPDLOG_ALL_HEADERS})
     target_compile_definitions(spdlog PUBLIC SPDLOG_SHARED_LIB)
-    if(MSVC)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
```

Comment:
This `if` statement is not necessary because `$<CXX_COMPILER_ID:MSVC>` appears in the following cmake-generator-expressions.

```cmake
        target_compile_options(spdlog PUBLIC $<$<AND:$<CXX_COMPILER_ID:MSVC>,$<NOT:$<COMPILE_LANGUAGE:CUDA>>>:/wd4251
                                             /wd4275>)
```

Reply:
> appears in the following cmake-generator-expressions.

but it is enabled only if build shared.

Reply:
> but it is reached only if build shared.

I overlooked it. In that sense, we need to set the same option for `spdlog::spdlog_header_only`.

Reply:
perhaps, but it is not directly related to this pr, so i will approve for now